mne/decoding/ssd.py

```diff
@@ -278,10 +282,6 @@ def apply(self, X):
             The processed data.
         """
         X_ssd = self.transform(X)
-        sorter_spec = Ellipsis
-        if self.sort_by_spectral_ratio:
-            _, sorter_spec = self.get_spectral_ratio(ssd_sources=X_ssd)
-
-        pick_patterns = self.patterns_[sorter_spec, :self.n_components].T
+        pick_patterns = self.patterns_[self.sorter_spec][:self.n_components].T
```

Comment:
why change the square brackets? Wouldn't the old brackets have worked after changing for self.sorter_spec ?

Reply:
the problem here was that since the first dimension is where the components are, we need first to sort them and then pick the components. I noticed this when testing `ssd.apply` with `n_components ! = n_channels`. In addition, it goes along how the `ssd.transform` is implemented (line 215/217)